src/main/java/org/zalando/problem/spring/web/advice/AdviceTrait.java

```diff
@@ -157,11 +157,15 @@ default boolean isCausalChainsEnabled() {
                     .headers(headers)
                     .contentType(contentType)
                     .body(problem);
-        }).orElseGet(() ->
-                ResponseEntity.status(HttpStatus.NOT_ACCEPTABLE).body(null)));
+        }).orElseGet(() -> fallback(throwable, problem, request, headers)));
```

Comment:
This is a good question. I don't have a clear opinion about it. My god feeling is that it is better to process the final result, regardless if it is fallback or not. If methods are separated and one overrides fallback method and has some processing logic then it may lead to code duplication or coupling of these two methods.